crates/currency/src/amount.rs

```diff
@@ -73,9 +73,8 @@ mod conversions {
             Ok(Self::new(amount, currency_id))
         }
 
-        pub fn to_unsigned_fixed_point(&self) -> Result<UnsignedFixedPoint<T>, DispatchError> {
-            let unsigned_fixed_point = <T as pallet::Config>::UnsignedFixedPoint::from_inner(self.amount);
-            Ok(unsigned_fixed_point)
+        pub fn to_unsigned_fixed_point(&self) -> UnsignedFixedPoint<T> {
+            <T as pallet::Config>::UnsignedFixedPoint::from_inner(self.amount)
```

Comment:
I don't think this does what you want it to - it interprets `self.amount` as being e.g. `FixedU128`. Meaning if `self.amount` is 5, it returns the same as `UnsignedFixedPoint::from_float(5e-18)`. You probably want to use `UnsignedFixedPoint::checked_from_integer` here.

Overall you seem to be conflating the balance type (which is `u128`) with `FixedU128`, but they are very much not the same thing. To clarify, the `Balance` type of u128 has a precision of one planck, whereas `FixedU128` has a precision of 1e-18 planck (at the cost of reducing the maximum resprensentable value). In the rpc's it doesn't make sense to convert the `Amount` to `FixedU128` - just return the u128 from `Amount::amount` directly. You should only use FixedU128s when you need the fractional precision.

The fact that all tests pass despite this function not doing what it was intended to do is a bit worrisome, we should have a look if we can improve the tests

Reply:
The concrete type of `inner` is the same as `BalanceOf<Self>`, but the fixed point implementation implicitly assumes this value has the granularity defined by `Self::DIV`. See: https://docs.rs/sp-arithmetic/6.0.0/src/sp_arithmetic/fixed_point.rs.html#126

Reply:
Update after fixing: This did affect some tests but since those were also affected by the type change, I thought I just had to adjust the expect values to have more decimals. The latest commit reverts many of the changes to the tests, you can see that by checking the overall test changes made by this PR.

Reply:
As to why this not affect lending behaviour more broadly in other tests: `from_unsigned_fixed_point` and `to_unsigned_fixed_point` were only used in tests and the rpc APIs. I did add a test for `to` and `from` FixedPoint numbers now.